src/Mod/Sketcher/Gui/DrawSketchHandlerBSpline.h

```diff
@@ -395,7 +395,12 @@ class DrawSketchHandlerBSpline: public DrawSketchHandler
 
     QString getCrosshairCursorSVGName() const override
     {
-        return QString::fromLatin1("Sketcher_Pointer_Create_BSpline");
+        if (SketcherGui::DrawSketchHandlerBSpline::ConstrMethod == 1) {

```

Comment:
Usually it is recommended to write the enum name rather than just the number 1. So something like : 
`SketcherGui::DrawSketchHandlerBSpline::ConstrMethod == SketcherGui::DrawSketchHandlerBSpline::ConstrMethod::NameOfConstrMethode`

However I see that bspline DSH do not use this syntax through the file. It uses : 
`ConstrMethod == 1`
So I suggest using the same syntax as is currently used through the file, so that when someone wants to change it, it will be easier to find.

Reply:
This is an old DSH, but even at those times, at least an enum should have been used to avoid "magical numbers". These go against code readability.

Modernising this is not very difficult, as it only means creating an ```enum class``` and correcting all instances in the file. This would also prevent bugs due to implicit conversion.

This PR did not introduce the issue, but it has some such instances, which highlights the issue. Possible ways forward are:
1. Ask the author to correct the issue by introducing the enum class and correcting.
2. Taking the responsibility as reviewer/maintainer, merging it with the issue and then making a PR to correct the issue after merge.
3. Do nothing about it. Somebody else will eventually fix it.

I would not advise to make a separate PR correcting the issue now, before merge, as this will make this PR conflict and will put more burden on the author.

Please, take a decision. I will go along with any you choose.

Reply:
I would leave it like this, the whole DSH will have to be rewriten to implement tool settings support anyway.

Reply:
OK.